src/plugins/validate-semantic/validators/oas3/refs.js

```diff
@@ -6,26 +6,58 @@ export const validateOAS3RefsForRequestBodiesReferenceRequestBodyPositions = ()
         const value = node.node
         const ref = value.$ref
 
-        if(!ref) {
+        if (!ref) {
           return acc
         }
 
         const [, refPath = ""] = ref.split("#")
         const pathArr = refPath.split("/") || []
         const parentRefKey = pathArr.slice(-2)[0]
         const targetRefKey = pathArr.slice(-1)[0]
-        if(
-          targetRefKey !== "requestBody"
-          && parentRefKey !== "requestBody"
-          && parentRefKey !== "requestBodies"
-        ) {
+
+        // External references starting with http:// or https://
+        if (ref.startsWith("https://") || ref.startsWith("http://")){
+          return acc  
+        }
+
+        // External references starting with ./
+        if (ref.startsWith("./")){
+          return acc  
+        }
+
+        // Extensions are valid
+        if (ref.startsWith("#/") && (parentRefKey.startsWith("/x-") || targetRefKey.startsWith("/x-"))){
+          return acc  
+        }
+
+        // Edge cases
+        if (ref.endsWith("requestBody") && (ref.startsWith("#/paths") || ref.startsWith("#/components"))){
+          return acc  
+        }
+
+        // External references ending with yaml
+        if (ref.endsWith(".yaml")) {
+          return acc
+        }
+
+        // Starting with #/compontents/schemas is not allowed
+        if (ref.startsWith("#/components/schemas")) {
+          acc.push({
+            level: "error",
+            message: `requestBody $refs cannot point to '#/components/schemas/…', they must point to '#/components/requestBodies/…'`,
```

Comment:
@shockey it was my request to add a separate message for this case because it's more common than others. "... position where a requestBody can be legally placed" just doesn't seem a clear enough indication of what's wrong in this particular case. It's better to be specific than vague.

Better wording suggestions are welcome!

> we usually don't say what the user has done wrong in our error messages _and_ how to fix it — it's generally one or the other.

A [good error message](https://hackernoon.com/what-makes-a-good-error-710d02682a68) tells both what's wrong _and_ how to fix it. Some errors are simple enough so that either the "what's wrong" or "how to fix" part is enough. Others could use a more detailed explanation. Fyi I also used the "what's wrong _and_ how to fix" pattern for the error messages [here](https://github.com/swagger-api/swagger-editor/pull/2105) and [here](https://github.com/swagger-api/swagger-editor/pull/2108).

Reply:
I left the message as we discussed with Helen, if there's a better message or makes sense let it as it was, i can do it.